src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -625,5 +626,112 @@ public override async Task<VSInternalWrapWithTagResponse> RazorWrapWithTagAsync(
+                                Methods.TextDocumentFoldingRange.Name,
+                                RazorLSPConstants.RazorCSharpLanguageServerName,
+                                SupportsFoldingRange,
+                                csharpRequestParams,
+                                cancellationToken).ConfigureAwait(false);
+
+                            var result = request.Result;
+                            if (result is null)
+                            {
+                                throw new InvalidOperationException("CSharp result is null");
+                            }
+
+                            csharpRanges.AddRange(result);
+                        }
+                    });
+
+            }
+
+            var htmlDocument = GetHtmlDocumentSnapshsot(foldingRangeParams.TextDocument.Uri.ToUri());
+            var htmlRanges = new List<OmniSharp.Extensions.LanguageServer.Protocol.Models.FoldingRange>();
+            var htmlTask = Task.CompletedTask;
+            if (htmlDocument is not null)
+            {
+                htmlTask = Task.Run(async () =>
+                    {
+                        var synchronized = await _documentSynchronizer.TrySynchronizeVirtualDocumentAsync(
+                        foldingRangeParams.HostDocumentVersion, htmlDocument, cancellationToken);
+
+                        if (synchronized)
+                        {
+                            var htmlRequestParams = new FoldingRangeParams()
+                            {
+                                TextDocument = new()
+                                {
+                                    Uri = htmlDocument.Uri
+                                }
+                            };
+
+                            var request = await _requestInvoker.ReinvokeRequestOnServerAsync<FoldingRangeParams, OmniSharp.Extensions.LanguageServer.Protocol.Models.FoldingRange[]>(
+                                Methods.TextDocumentFoldingRange.Name,
+                                RazorLSPConstants.HtmlLanguageServerName,
+                                SupportsFoldingRange,
+                                htmlRequestParams,
+                                cancellationToken).ConfigureAwait(false);
+
+                            var result = request.Result;
+                            if (result is null)
+                            {
+                                throw new InvalidOperationException("Html result is null");
+                            }
+
+                            htmlRanges.AddRange(result);
+                        }
+                    });
+            }
+
+            // Return null if any of the tasks getting folding ranges
+            // results in an error
+            var allTasks = Task.WhenAll(htmlTask, csharpTask);
+            await allTasks.ConfigureAwait(false);
+            if (allTasks.IsFaulted)
```

Comment:
Wont the above `await` throw if it faults?

Reply:
I'm not actually sure... I meant to ask/lookup :)

Reply:
It does throw. Fixing